pgjdbc/src/main/java/org/postgresql/jdbc/ArrayEncoding.java

```diff
@@ -1076,6 +1076,8 @@ public void appendArray(StringBuilder sb, char delim, Object[] array) {
             // this should never happen
             throw new IllegalStateException(e);
           }
+        } else if (array[i] instanceof java.sql.Array) {
+          sb.append(array[i]);
```

Comment:
What is the toString implementation of java.sql.Array?

Reply:
Currently only PgArray implements Array on the codebase so it just returns the fieldString of PgArray, do you prefer checking with **instanceof PgArray**?
The intention behind this line is to prevent the code from falling to the else block which also escaped Arrays, which we shouldn't do.

Reply:
An array of Arrays has never been supported (as best I can tell), and it is not clear we should support it.
But if we are going to support, it should be done as its own feature/pr.

Reply:
https://github.com/pgjdbc/pgjdbc/pull/3337